wordpress-mobile/WordPress-Android
# Consolidate tracking onTabChanged in the main Reader view

Parent issue #12028

I recommend reviewing this PR by commits.

Goal of this PR is to correctly track TabShown events in the main Reader view.
- Following tab - `reader_following_shown`
- Discover tab - `reader_discover_shown`
- Liked tab - `reader_liked_shown`
- Custom tab (available only on old .com accounts) - `reader_custom_tab_shown`
- Saved tab - `reader_saved_list_shown`
                  - `source` property `reader_filter` - tab in the main Reader view
                  - `source` property `post_list_saved_post_notice` - action in snackbar shown after the user bookmarks a post in the main Reader view
                  - `source` property `post_details_saved_post_notice` - action in snackbar shown after the user bookmarks a post in post detail view


The events are tracked when the user navigates to the tab
- When the user opens the main Reader view from the bottom navigation bar
- When the app comes from background and the main Reader view is shown
- When the user changes tab in the main Reader view

The events are NOT tracked when
- the device orientation changes
- user opens eg. post detail and they come back to the main Reader view


To test - enable RI FF
Verify that the events are tracked as per the description above.
+
1. Open Reader - discover tab
2. Kill the app
3. Open the app and notice only "reader_discover_shown" is tracked


PR submission checklist:
